randomwallpaper@iflow.space/wallpaperController.js

```diff
@@ -403,21 +404,24 @@ var WallpaperController = class {
 		});
 	}
 
-	_getGeneralPostCommand(historyElement) {
-		let commandString = this._settings.get('general-post-command', 'string');
-
-		if (commandString === "") {
+	_getCommandArray(commandString, historyElement) {
+		let string = commandString;
+		if (string === "") {
 			return null;
 		}
 
-		let commandArray = commandString.split(' ');
+		// Replace variables
+		let variables = new Map();
+		variables.set('%wallpaper_path%', historyElement.path);
 
-		commandArray.forEach((value, index, array) => {
-			if (value === "%wallpaper_path%") {
-				array[index] = historyElement.path;
-			}
+		variables.forEach((value, key) => {
+			string = string.replaceAll(key, value);
 		});
 
+		// https://stackoverflow.com/a/43766456
+		const regex = /("[^"\\]*(?:\\[\S\s][^"\\]*)*"|'[^'\\]*(?:\\[\S\s][^'\\]*)*'|\/[^\/\\]*(?:\\[\S\s][^\/\\]*)*\/[gimy]*(?=\s|$)|(?:\\\s|\S)+)/g;
```

Comment:
Phu, this is a hefty regex, but glad you figured that splitting by space isn't enough.

I think in our case, we could simplify `[\S\s]` with just `.`, because we probably don't need to support commands with new lines but that doesn't make the regex a lot easier to read. So let's just keep it and never look at it again if we don't have to :sweat_smile: ...

Reply:
This will be adjusted in a later commit. I discovered that GLib can do this for us.
[`d9ecd7b` (#154)](https://github.com/ifl0w/RandomWallpaperGnome3/pull/154/commits/d9ecd7b2b5ecba7bff4dc95668087c05c326360a)

Sorry again for these things.

Reply:
A that's even better! Offload that burden of parsing to someone else! :D 

If you want to take a look at [their implementation](https://github.com/GNOME/glib/blob/d9a01931c3dd34ab07b7ac18058f1b805a7d76f4/glib/gshell.c#L661) they do not use any fancy regex.